From 254f641f9cea15c476432860f322d2b810bda679 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <matt.ranostay@intel.com>
Date: Fri, 5 Feb 2016 14:32:00 -0800
Subject: [PATCH 2/3] usb: gadget: android: fix deference after free

Error pointer config->f_midi_inst was dereferenced after config
objects is kfree'ed.

Change-Id: Idb6da420183990cdf2c0d34e340a60c3fe28cdb2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18396
Signed-off-by: Matt Ranostay <matt.ranostay@intel.com>
Reviewed-on: https://android.intel.com:443/467671
---
 drivers/usb/gadget/android.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index b4aecc5..7f94248 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -1142,8 +1142,9 @@ static int midi_function_init(struct android_usb_function *f,
 
 	config->f_midi_inst = usb_get_function_instance("midi");
 	if (IS_ERR(config->f_midi_inst)) {
+		int ret = PTR_ERR(config->f_midi_inst);
 		kfree(config);
-		return PTR_ERR(config->f_midi_inst);
+		return ret;
 	}
 
 	midi_opts = container_of(config->f_midi_inst, struct f_midi_opts,
-- 
1.9.1

